lib/seahorn/BvOpsem2WideMemMgr.cc

```diff
@@ -0,0 +1,543 @@
+#include "BvOpSem2Context.hh"
+#include "BvOpSem2RawMemMgr.hh"
+
+#include "BvOpSem2MemManagerMixin.hh"
+
+#include "seahorn/Expr/ExprOpStruct.hh"
+#include "seahorn/Support/SeaDebug.h"
+#include "seahorn/Support/SeaLog.hh"
+
+static const unsigned int g_slotBitWidth = 64;
+static const unsigned int g_slotByteWidth = g_slotBitWidth / 8;
+
+static const unsigned int g_uninit = 0xDEADBEEF;
+
+namespace seahorn {
+namespace details {
+class WideMemManager : public OpSemMemManagerBase {
+
+  /// \brief Allocates memory regions in virtual memory
+  std::unique_ptr<OpSemAllocator> m_allocator;
+
+  /// \brief Knows the memory representation and how to access it
+  std::unique_ptr<OpSemMemRepr> m_memRepr;
+
+  /// \brief Base name for non-deterministic pointer
+  Expr m_freshPtrName;
+
+  /// \brief Register that contains the value of the stack pointer on
+  /// function entry
+  Expr m_sp0;
+
+  /// \brief Source of unique identifiers
+  mutable unsigned m_id;
+
+  const Expr m_uninit_size = m_ctx.alu().si(g_uninit, g_slotBitWidth);
```

Comment:
I am never clear about the rules when such initialization are ok. They were not ok before in C++ and now they are sometimes ok. I would prefer for this to be moved to the constructor.

Reply:
ack.

Reply:
why can't `m_unit_size` be initialized in the initializer list right before `m_nullptr`?

Reply:
Good point. done.